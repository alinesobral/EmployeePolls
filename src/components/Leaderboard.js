import { connect } from "react-redux";

const Leaderboard = (props) => {
  console.log("Leaderboard props: ", props.users);
  return (
    <div>
      <h1>Leaderboard</h1>
      <table>
        <thead>
          <tr>
            <th>Users</th>
            <th>Answered</th>
            <th>Created</th>
          </tr>
        </thead>
        <tbody>
          {props.users.map((user) => (
            <tr key={user.id}>
              <td className="wide">
                <div>
                  <img src={user.avatarURL} alt="user's avatar" />
                </div>
                <div>{user.name} </div>
                <div className="user-handle">{user.id}</div>
              </td>
              <td>
                <span>{Object.keys(user.answers).length}</span>
              </td>
              <td>
                <span>{user.questions.length}</span>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

const mapStateToProps = ({ users }) => {
  return {
    users: Object.values(users).sort(
      (a, b) =>
        Object.keys(b.answers).length +
        b.questions.length -
        (Object.keys(a.answers).length + a.questions.length)
    ),
  };
};

export default connect(mapStateToProps)(Leaderboard);
