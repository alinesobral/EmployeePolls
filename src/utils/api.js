import {
  _saveQuestion,
  _saveQuestionAnswer,
  _getUsers,
  _getQuestions,
  _login,
  _logout,
} from "./_DATA";

export function getInitialData() {
  return Promise.all([_getUsers(), _getQuestions()]).then(
    ([users, questions]) => ({
      users,
      questions,
    })
  );
}

/**
 * question: {
 *        optionOneText,
 *        optionTwoText,
 *        author
 * }
 */
export function saveQuestion(question) {
  return _saveQuestion(question);
}

/**
 * info : {
 *        authedUser,
 *        qid,
 *        answer,
 * }
 * */
export function saveQuestionAnswer(info) {
  return _saveQuestionAnswer(info);
}

export function login(user) {
  return _login(user);
}

export function logout() {
  return _logout();
}
