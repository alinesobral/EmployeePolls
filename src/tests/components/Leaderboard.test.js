import { Provider } from "react-redux";
import { createStore } from "redux";
import { MemoryRouter, BrowserRouter } from "react-router-dom";
import { render, screen, fireEvent } from "@testing-library/react";
import { _login } from "../../utils/_DATA";
import userEvent from "@testing-library/user-event";

import reducer from "../../reducers";
import middleware from "../../middleware";
import Leaderboard from "../../components/Leaderboard";
// import { store } from "../../index.js";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureStore([thunk]);

describe("Leaderboard", () => {
  let store, component;

  beforeEach(() => {
    store = mockStore({
      authedUser: "sarahedo",
    });

    component = render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>
    );
  });

  it("will show the leaderboard page", () => {
    // console.log(store);
    // var component = ReactDOM.render(
    //   <Provider store={store}>
    //     <MemoryRouter>
    //       <Leaderboard />
    //     </MemoryRouter>
    //   </Provider>
    // );
    // expect(component).toMatchSnapshot();
  });
});
