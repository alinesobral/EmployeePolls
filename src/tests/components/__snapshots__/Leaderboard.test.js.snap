// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard will show the leaderboard page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          Leaderboard
        </h1>
        <table>
          <thead>
            <tr>
              <th>
                Users
              </th>
              <th>
                Answered
              </th>
              <th>
                Created
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="wide"
              >
                <div>
                  <img
                    alt="user's avatar"
                    class="leaderboard-avatar"
                    src="/images/sarahedo.jpg"
                  />
                </div>
                <div>
                  Sarah Edo
                   
                </div>
                <div
                  class="user-handle"
                >
                  sarahedo
                </div>
              </td>
              <td>
                <span>
                  4
                </span>
              </td>
              <td>
                <span>
                  2
                </span>
              </td>
            </tr>
            <tr>
              <td
                class="wide"
              >
                <div>
                  <img
                    alt="user's avatar"
                    class="leaderboard-avatar"
                    src="images/"
                  />
                </div>
                <div>
                  Tyler McGinnis
                   
                </div>
                <div
                  class="user-handle"
                >
                  tylermcginnis
                </div>
              </td>
              <td>
                <span>
                  2
                </span>
              </td>
              <td>
                <span>
                  2
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        Leaderboard
      </h1>
      <table>
        <thead>
          <tr>
            <th>
              Users
            </th>
            <th>
              Answered
            </th>
            <th>
              Created
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="wide"
            >
              <div>
                <img
                  alt="user's avatar"
                  class="leaderboard-avatar"
                  src="/images/sarahedo.jpg"
                />
              </div>
              <div>
                Sarah Edo
                 
              </div>
              <div
                class="user-handle"
              >
                sarahedo
              </div>
            </td>
            <td>
              <span>
                4
              </span>
            </td>
            <td>
              <span>
                2
              </span>
            </td>
          </tr>
          <tr>
            <td
              class="wide"
            >
              <div>
                <img
                  alt="user's avatar"
                  class="leaderboard-avatar"
                  src="images/"
                />
              </div>
              <div>
                Tyler McGinnis
                 
              </div>
              <div
                class="user-handle"
              >
                tylermcginnis
              </div>
            </td>
            <td>
              <span>
                2
              </span>
            </td>
            <td>
              <span>
                2
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
